### DISCUSSION TITLE
_app.js: export default withAuthenticator(AppComponent) breaks

### DISCUSSION THREAD

**Discussion by paragbaxi** [vercel/next.js#17699] [Help]
I'd like to wrap my entire next.js app to be authenticated. The easiest way to do is to wrap [AWS Amplify's](https://docs.amplify.aws/lib/auth/getting-started/q/platform/js#option-1-use-pre-built-ui-components) `withAuthenticator` as a HoC to a custom `_app.js`. 

`export default withAuthenticator(AppComponent);`
unfortunately this produces an error

> Unhandled Runtime Error
> Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
> 
> Check the render method of AppComponent. 

I've traced back the error to `<Component {...pageProps} />`. I've searched Github and Google for anyone else that has examples of this design pattern already with no success.

This feels like I'm doing it wrong, which means I wrap withAuthenticator on every individual page–not necessarily bad, just a pain.
Curious to if anyone has any ideas.

**Comment by nickcorbett15** [comment]
Currently dealing with the same issue except I am not getting an error. 

If I test my app and go to localhost:3000 I have to log in but without logging in if I got to localhost:3000/dashboard
the dashboard component will load even if I am not logged in.
